backend/infrastructure/KeyVaultService.cs

```diff
@@ -80,4 +101,20 @@ public static async Task<string>  GetMqttToken()
 
         return mqttToken;
     }
+    
+    public static async Task<string> GetKeyType(string keyName)
+    {
+        var kvUri = "https://climatekv.vault.azure.net/";
+        var client = new KeyClient(new Uri(kvUri), new DefaultAzureCredential());
+        try
+        {
+            KeyVaultKey key = await client.GetKeyAsync(keyName);
+            return key.KeyType.ToString();
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine($"Error occurred while retrieving key from Azure Key Vault: {e.Message}");
+            return null;
+        }
```

Comment:
God brug af try catch til at hente fra key vault, men gør gerne brug af vores error handler istedet, kast en error istedet for at return null, da vores program.cs sørger for at printe fejlen, så vi kan slå det fra i et live miljø, eller logge det til en ekstern logger.